ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java

```diff
@@ -148,6 +148,18 @@ public void onCompletion(Long result, Exception exception) {
     if (!responseMetadataWriteInitiated.get()) {
       maybeWriteResponseMetadata(responseMetadata, new ResponseMetadataWriteListener());
     }
+    if (finalResponseMetadata == null) {
```

Comment:
this looks good, but can you add a test case for this logic?

Reply:
+1, I guess we can at least test the case where the channel becomes inactive